src/Apps/Order/Components/SavedAddresses.tsx

```diff
@@ -1,90 +1,101 @@
-import { AddressModal, ModalDetails } from "Apps/Order/Components/AddressModal"
-import { CommitMutation } from "Apps/Order/Utils/commitMutation"
+import {
+  AddressModal,
+  AddressModalAction,
+  AddressModalActionType,
+} from "Apps/Order/Components/AddressModal"
 import createLogger from "Utils/logger"
 import { SavedAddressItem } from "Apps/Order/Components/SavedAddressItem"
 import { deleteUserAddress } from "Apps/Order/Mutations/DeleteUserAddress"
 import { useSystemContext } from "System/SystemContext"
-import { compact } from "lodash"
 import { extractNodes } from "Utils/extractNodes"
-import { UpdateUserAddressMutation$data } from "__generated__/UpdateUserAddressMutation.graphql"
-import { CreateUserAddressMutation$data } from "__generated__/CreateUserAddressMutation.graphql"
 import { useTracking } from "react-tracking"
 import { ActionType, ContextModule, OwnerType } from "@artsy/cohesion"
 import styled from "styled-components"
 import { themeGet } from "@styled-system/theme-get"
+import { SavedAddressType } from "Components/Address/utils"
+import {
+  ShippingAddressFormValues,
+  addressWithFallbackValues,
+  getDefaultUserAddress,
+} from "Apps/Order/Utils/shippingUtils"
+import { compact } from "lodash"
 
 export const NEW_ADDRESS = "NEW_ADDRESS"
 const PAGE_SIZE = 30
 
-interface SavedAddressesProps {
-  onChangeAddressCount?: (active?: number) => void
-  me: SavedAddresses_me$data
-  onSelect?: (string) => void
-  commitMutation?: CommitMutation
+export interface SavedAddressesProps {
   relay: RelayRefetchProp
-  addressCount?: number
-  onAddressDelete?: (removedAddressId: string) => void
-  onAddressCreate?: (
-    address: CreateUserAddressMutation$data["createUserAddress"]
-  ) => void
-  onAddressEdit?: (
-    address: UpdateUserAddressMutation$data["updateUserAddress"]
-  ) => void
-  selectedAddress?: string
-  onShowToast?: (isShow: boolean, action: string) => void
+  me: SavedAddresses_me$data
+  active: boolean
+  onSelect: (address: ShippingAddressFormValues) => void
 }
 
-type Address = NonNullable<
-  NonNullable<
-    NonNullable<
-      NonNullable<SavedAddresses_me$data["addressConnection"]>["edges"]
-    >[0]
-  >["node"]
->
-
-const defaultAddressIndex = (addressList: Address[]) => {
-  const items = compact(addressList)
-
-  if (!items || items.length == 0) {
-    return
-  }
-
-  const defaultAddressID = items.find(node => node.isDefault)?.internalID
+const getAddressByID = (addressList: SavedAddressType[], addressID: string) => {
+  return addressList.find(node => node.internalID === addressID)
+}
 
-  return defaultAddressID || items[0].internalID
+const getBestAvailableAddress = (
+  addressList: SavedAddressType[],
+  addressID?: string
+) => {
+  return (
+    (addressID && getAddressByID(addressList, addressID)) ||
+    getDefaultUserAddress(addressList)
+  )
 }
 
 const SavedAddresses: React.FC<SavedAddressesProps> = props => {
-  const { trackEvent } = useTracking()
```

Comment:
SavedAddresses, SavedAddressItem and AddressModal are responsible for creating, editing and deleting user addresses. In addition when a user selects an address it sets all address values in the FulfillmentDetailsForm formik context.

Reply:
**question:** I didn't expect this refactor to affect saved addresses. Does FulfillmentDetailsForm need to know about saved addresses because it is now responsible for submitting the fulfillment details regardless of whether they are new or existing?

Reply:
Yes: Considering fulfillment details as a step separate from the shipping quotes, the `SavedAddresses` component acts as a single input to set the entire address object. _Inside_ the component are controls to create/update/delete saved addresses, which in turn affect the available options for the `FulfillmentDetailsForm`. This is how it worked before, just with that logic localized with its behavior in a more predictable way.